### ISSUE TITLE
Issue with pre-built binaries needs a new release of Carthage

### ISSUE THREAD

**Issue by sam-w** [Carthage/Carthage#2986]
* carthage install method: [x] .pkg, [x] homebrew, [] source
* `which carthage`: all of them
* `carthage version`: 0.34.0
* `xcodebuild -version`: 11.4

**Actual outcome**
Carthage has not released a new version since Oct 2019.

**Expected outcome**
Carthage should release a new version ASAP.

## Background

Hi all, as referenced in https://github.com/Carthage/Carthage/issues/2887, there is an issue with Carthage which causes it to fail builds because the local Swift version doesn't match the framework Swift version _even_ if both are module stable.

## Impact

**This is a blocker to Carthage adoption** if your project uses a dependency with a pre-built binary, for example:
- [Instabug](https://github.com/Instabug/Instabug-iOS/issues/364)
- [Couchbase](https://github.com/couchbase/couchbase-lite-ios/issues/2570)
- [Topee](https://github.com/avast/topee/issues/13)
- [Onfido](https://github.com/onfido/onfido-ios-sdk/issues/124)
- [Bugfender](https://github.com/bugfender/BugfenderSDK-iOS/issues/54)
- [Phrase](https://github.com/phrase/ios-sdk/issues/13)

Workarounds exist for some dependencies with pre-built binaries, but many SDK providers do not support an alternative. Developers are forced instead to find an alternative to Carthage.

The impact is not only on developers. Maintainers of the binary-only dependencies are forced to spend time investigating (and apologising for) a Carthage issue.

## Solution

[This has already been solved!](https://github.com/Carthage/Carthage/pull/2902). AFAIK, all that is required is a new release of Carthage.

At the very least, given the reliance of the industry on Carthage and the ongoing inconvenience, we would appreciate an explanation of why a release is not forthcoming and when we might expect one.

Thanks for all you do. <3

**Comment by tmspzz** [comment]
I'll close this as a duplicate of https://github.com/Carthage/Carthage/issues/2927 . The reason is pretty much explained here https://github.com/Carthage/Carthage/pull/2972#issuecomment-613597401.

In the mean time, you can build Carthage from source. As you have pointed out the problem does not exist on master.

**Comment by elliottwilliams** [comment]
@tmspzz I must be missing something. Why can't this be released?

In https://github.com/Carthage/Carthage/pull/2972#issuecomment-613597401, I was talking about why Carthage can't adopt Swift 5 / Xcode 11-only features. But #2902 has already been merged to master, and master builds and passes tests on High Sierra / Xcode 10.1. Is there something else in the way? I'd be happy to work on unblocking this + other changes that folks want to see in a release.

**Comment by sam-w** [comment]
@tmspzz please consider reopening this issue given what @elliottwilliams is saying.

> In the mean time, you can build Carthage from source.

Yes, I can (and am). That's not a scalable solution though.

Could you also help me understand the resistance to cutting a new release from the point-of-view of dropping High Sierra support?

Like @elliottwilliams mentions above:

> master builds and passes tests on High Sierra / Xcode 10.1

... but even if this weren't true, it's not like dropping High Sierra support would have much impact. You guys aren't releasing anything right now! Seems like paralysing logic to me!

Again, thanks for all you do and for giving this issue some more consideration.

**Comment by elliottwilliams** [comment]
To be clear, I'm not advocating dropping high sierra support without a plan for the impacted users. I'm just curious whether it's possible to get a release out in the meantime. If not, then it sounds like Carthage really _is_ blocked on dropping high sierra and we'll have to work on that before a new release can be cut.

**Comment by ikesyo** [comment]
At the time of May 2020, App Store requires submitted apps to be built using Xcode 11, which requires macOS Mojave 10.14.3 or later. That means that we do not need to support High Sierra anymore.

https://developer.apple.com/app-store/submissions/

> ## Build with Xcode 11.
> Xcode 11 includes Swift 5 and the latest SDKs for all Apple platforms. With Swift 5, runtime is built into iOS, iPadOS, macOS, tvOS, and watchOS so your app no longer needs to bundle this library for these latest OS releases. Users who download an app built with Swift from the App Store will get a smaller app and a faster download.
> 
> Starting April, 2020, all apps submitted to the App Store will need to be built with Xcode 11. Xcode 11 requires macOS Mojave 10.14.3 or later.

**Comment by jdhealy** [comment]
Great High Sierra content for [the High Sierra conversation](https://github.com/Carthage/Carthage/pull/2972#issuecomment-617308464), we can continue that in #2972

As for evaluating the releasability of [what’s currently on master](https://github.com/Carthage/Carthage/tree/f656edfe35651b54eec50d814e79d079f8eea7c4), first the hold up was [fixing breakage of the `--project-directory` flag](https://github.com/Carthage/Carthage/commit/f656edfe35651b54eec50d814e79d079f8eea7c4)…

**Comment by jdhealy** [comment]
More recently, the hold up has been finding time for the trio of through release notes on breaking changes, time to assemble the PKG package, and (most crucially) evaluating recent Homebrew bottling changes (extremely important because that’s assumed — as we don’t collect stats — to be how Carthage consumers — and some CI systems — rely upon installation).
